libass/ass_font.c

```diff
@@ -95,14 +95,28 @@ uint32_t ass_font_index_magic(FT_Face face, uint32_t symbol)
     }
 }
 
-static void buggy_font_workaround(FT_Face face)
+static void set_font_metrics(FT_Face face)
 {
-    // Some fonts have zero Ascender/Descender fields in 'hhea' table.
-    // In this case, get the information from 'os2' table or, as
-    // a last resort, from face.bbox.
+    // Mimicking GDI's behavior for asc/desc/height
+    TT_OS2 *os2 = FT_Get_Sfnt_Table(face, ft_sfnt_os2);
+    if (os2 && (os2->usWinAscent + os2->usWinDescent != 0)) {
```

Comment:
I think this needs extra `(short)` as mentioned in a comment that is removed by this commit (if we are to believe that comment). The assignments a few lines below don’t need it because they assign to `short` (although this goes through implementation-defined behaviour, but so does `(short)`), but this line promotes the `unsigned short`s to `int` during the addition, so the sum will never be zero unless both values are zero.

Reply:
Oh, totally overlooked that comment.
I'm adding the casts on the assignments as well for the sake of clarity, along with a corresponding comment.